### PR TITLE
[msi] release new api 2018-11-30 version

### PR DESCRIPTION
**Pull request by LucasYao93**
## Description

+ add test cases.
+ add examples and docs.

Design: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1129

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by LucasYao93** [comment]
We need to provide engineering builds for the service team.

File: src/ManagedServiceIdentity/docs/Update-AzUserAssignedIdentity.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+## SYNOPSIS
+Update an identity in the specified subscription and resource group.
+
+## SYNTAX
+
+### UpdateExpanded (Default)
+```
+Update-AzUserAssignedIdentity -Name <String> -ResourceGroupName <String> [-SubscriptionId <String>]
+ [-Location <String>] [-Tag <Hashtable>] [-DefaultProfile <PSObject>] [-Confirm] [-WhatIf]
```
Comment (BethanyZhou) [inline src/ManagedServiceIdentity/docs/Update-AzUserAssignedIdentity.md:18]: I suspect Location is required for `Update-AzUserAssignedIdentity`, could you verify it?

**Comment by BethanyZhou** [comment]
Please submit a design review.

File: src/ManagedServiceIdentity/docs/Update-AzUserAssignedIdentity.md (hunk 1 above)
Comment (LucasYao93) [reply src/ManagedServiceIdentity/docs/Update-AzUserAssignedIdentity.md:18]: I have verified that the Location is optional for update operation.

**Comment by LucasYao93** [comment]
> Please submit a design review.

https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1129

**Comment by BethanyZhou** [comment]
> > Please submit a design review.
> 
> [Azure/azure-powershell-cmdlet-review-pr#1129](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1129)

Thanks @LucasYao93 , please link it to Checklist in next time

**Comment by BethanyZhou** [comment]
@LucasYao93  Current PR looks good to me. Was this engineer build verified by service team?  If so, we can consider merge it. And could you please provide a changelog message for your changes?

**Comment by LucasYao93** [comment]
Hi @BethanyZhou,

Please merge the PR.

For merge into main branch.
The old resource base on SDK https://github.com/Azure/azure-powershell/tree/main/src/ManagedServiceIdentity. we use generation module replace it. So first we should delete it on the main branch.
What should the delete step do?

Thanks

**Comment by BethanyZhou** [comment]
Is there any breaking change will be introduced?

**Review by BethanyZhou — APPROVED** [review]

**Comment by LucasYao93** [comment]
> Is there any breaking change will be introduced?
@BethanyZhou ,
The old resource module version is 0.7.3(https://www.powershellgallery.com/packages/Az.ManagedServiceIdentity/0.7.3).
we don't wait major release. please merge it and then trigger code/gen pipeline.

Thanks

**Pull request merged**